pkg/reconciler/common/releases.go

```diff
@@ -82,6 +82,12 @@ func InstalledManifest(instance v1alpha1.KComponent) (mf.Manifest, error) {
 func IsVersionValidMigrationEligible(instance v1alpha1.KComponent) error {
 	var err error
 	target := sanitizeSemver(TargetVersion(instance))
+	if target == "v" {
+		// If the function TargetVersion returns empty string, if means the specified target version is not available
+		// in this release to install.
+		return fmt.Errorf("target version %v is not available in this release.", instance.GetSpec().GetVersion())
+	}
```

Comment:
May I suggest this change to make things more readable? It feels to me like we should fail immediately after calling `TargetVersion()`, instead of passing something invalid to `sanitizeSemver()`, since we already know it's invalid at that point.

```suggestion
	trgtVer := TargetVersion(instance)
	if trgtVer == "" {
		// If the function TargetVersion returns empty string, if means the specified target version is not available
		// in this release to install.
		return fmt.Errorf("target version %v is not available in this release.", instance.GetSpec().GetVersion())
	}

	target := sanitizeSemver(trgtVer)
```